### ISSUE TITLE
SpectralCoord sometimes inverts RV on initialization

### ISSUE THREAD

**Issue by rosteen** [astropy/specutils#646]
@eteq @nmearl A more detailed report of what I brought up privately today:

If a target and radial_velocity, but no observer, are input to SpectralCoord, the resulting object has the inverse of the supplied radial_velocity. If you then create a new SpectralCoord object from the first, the radial_velocity value flips again. I would have caught this sooner, but had a couple erroneous lines of `assert spec_coord2.[param] == spec_coord2.[param]` instead of `assert spec_coord1.[param] == spec_coord2.[param]` in test_spectral_coord.test_create_from_spectral_coord. See the example below:

```
>>> spec_coord1 = SpectralCoord([100, 200, 300] * u.nm, observer = None, target = LSRD_DIR_STATIONARY_EQUIV[0], radial_velocity = u.Quantity(500, 'km/s'), doppler_convention = 'optical', doppler_rest = 6000*u.AA)
>>> spec_coord1
<SpectralCoord [100., 200., 300.] nm,
	radial_velocity=-500.0 km / s,
	redshift=-0.00166782047599076,
	doppler_rest=6000.0 Angstrom,
	doppler_convention=optical,
	observer=None,
	target=Galactic>
>>> spec_coord2 = SpectralCoord(spec_coord1)
>>> spec_coord2
<SpectralCoord [100., 200., 300.] nm,
	radial_velocity=500.0 km / s,
	redshift=0.00166782047599076,
	doppler_rest=6000.0 Angstrom,
	doppler_convention=optical,
	observer=None,
	target=Galactic>
>>>
>>>
>>>
>>> spec_coord3 = SpectralCoord(spec_coord2)
>>> spec_coord3
<SpectralCoord [100., 200., 300.] nm,
	radial_velocity=-500.0 km / s,
	redshift=-0.00166782047599076,
	doppler_rest=6000.0 Angstrom,
	doppler_convention=optical,
	observer=None,
	target=Galactic>
```

**Comment by rosteen** [comment]
Upon some further digging, I think I see what's happening here. In this case, a private observer (`self._observer`) is getting created with `self._target_from_observer(target, -radial_velocity)`. This isn't available to the user via `spec_coord.observer` because `self._frames_state['observer']` is still False, so self.observer returns None. That private variable `self._observer` _is_, however, being used in the radial_velocity property, which is defined by a calculation using `self._target` and `self._observer`, which in this case results in the inverse of the radial_velocity originally supplied by the user. 

Considering the input RV is also ignored if both observer and target are input, perhaps the solution here is a Warning or Error about the effects of creating a SpectralCoord with both target/observer and radial_velocity inputs.

**Comment by nmearl** [comment]
> Upon some further digging, I think I see what's happening here. In this case, a private observer (self._observer) is getting created with self._target_from_observer(target, -radial_velocity). This isn't available to the user via spec_coord.observer because self._frames_state['observer'] is still False, so self.observer returns None. That private variable self._observer is, however, being used in the radial_velocity property, which is defined by a calculation using self._target and self._observer, which in this case results in the inverse of the radial_velocity originally supplied by the user.

Correct. An internal `observer` is created to keep all mathematical transformations consistent. As such, `radial_velocity` and `redshift` don't actually _exist_; instead, they are calculated from the relation between an `observer`/`target` pair. If either the `observer` or `target` doesn't exist, then an implicit one is created "under the hood" to preserve the radial velocity / redshift.

The way this is done is given e.g. an `observer`, a `target` is created with a velocity offset equal to the radial velocity along the positive `x` axis in a cartesians ICRS frame. Likewise, given a `target` and no `observer` creates an observer along the negative `x` axis (`-radial_velocity`, hence the formalism).

There used to be a reason for this: the vector calculates required sign preservation. But when I look back through the code, I think I may have solved this particular issue because I'm not seeing any immediate consequences of just removed the leading `-` on line 118. All tests seem to pass as well. 🤷‍♂

**Comment by rosteen** [comment]
Well, if the reason this was negative was due to calculation constraints (not a scientific convention) and it works without the negative - I'll just ditch the negative sign and close this? I just tested and confirmed that fixes what I was seeing, and doesn't cause any tests to fail.

**Comment by nmearl** [comment]
Sound good! We'll just keep this issue in mind if an error pops up in the future. 😉

**Comment by eteq** [comment]
@nmearl @rosteen - I'm confused - was the conclusion here "won't fix"?  this seems to be a pretty serious problem as originally posed - it's definitely not right if the sign of the RV flips just on copy...

Put another way, it's definitely wrong if the user explicitly gives a positive RV and it spontaneously changes to negative without them asking for that.

**Comment by nmearl** [comment]
It is fixed in #645.
